test/unit/messengers/cinstance_messenger_test.rb

```diff
@@ -183,7 +183,8 @@ def test_plan_change
       assert messenger.deliver
       assert provider.received_messages.empty?
 
-      message = master.received_messages.last!
+      # Using first! as they are ordered by created_at desc
+      message = master.received_messages.first!
```

Comment:
![image](https://user-images.githubusercontent.com/11318903/74237236-5c62a300-4cd3-11ea-801a-4f424df00bca.png)

result of `bundle exec rake test TEST=test/unit/messengers/cinstance_messenger_test.rb TESTOPS="--name= test_plan_change -v"`:
![image](https://user-images.githubusercontent.com/11318903/74237287-7b613500-4cd3-11ea-8a2b-b9dd5ac0d783.png)

And it is true that the `created_at` does not exist for that table, so this comment is wrong.
https://github.com/3scale/porta/blob/e2dbb0cf87ed7a340261f681b9444254dd1a199b/db/schema.rb#L799-L809

And if right before this line I add a `pp master.received_messages.count`, there is only 1 (with mysql, I didn't try with other DBs) so I do not understand why this change but I assume it is for other DBs.

However, let's remember that for Postgres the order is 'random' and that there is ever more than 1, we will get random failures in circleci... so we should set the order manually for example with `message = master.received_messages.order(id: :desc).first!` in order to don't drive us too crazy if we ever see a circleci error in postgres here.

Reply:
Order is ensured by https://github.com/3scale/porta/blob/e2dbb0cf87ed7a340261f681b9444254dd1a199b/app/models/account.rb#L166

Try in the console `MessageRecipient.latest_first`. It uses the `Message`'s `created_at`.

Reply:
Ahaaa, I see :+1: Sorry :+1: Then imo, we should either remove that message or mention that it is the created_at of its message and not its own created_at :smile: Anyway, it is clear now that it isn't a bug so thanks :+1: